src/linalg/inverse.rs

```diff
@@ -145,124 +145,149 @@ where
-
-    out[(0, 2)] = m[4].clone() * m[9].clone() * m[15].clone()
-        - m[4].clone() * m[11].clone() * m[13].clone()
-        - m[8].clone() * m[5].clone() * m[15].clone()
-        + m[8].clone() * m[7].clone() * m[13].clone()
-        + m[12].clone() * m[5].clone() * m[11].clone()
-        - m[12].clone() * m[7].clone() * m[9].clone();
-
-    out[(1, 2)] = -m[0].clone() * m[9].clone() * m[15].clone()
-        + m[0].clone() * m[11].clone() * m[13].clone()
-        + m[8].clone() * m[1].clone() * m[15].clone()
-        - m[8].clone() * m[3].clone() * m[13].clone()
-        - m[12].clone() * m[1].clone() * m[11].clone()
-        + m[12].clone() * m[3].clone() * m[9].clone();
-
-    out[(2, 2)] = m[0].clone() * m[5].clone() * m[15].clone()
-        - m[0].clone() * m[7].clone() * m[13].clone()
-        - m[4].clone() * m[1].clone() * m[15].clone()
-        + m[4].clone() * m[3].clone() * m[13].clone()
-        + m[12].clone() * m[1].clone() * m[7].clone()
-        - m[12].clone() * m[3].clone() * m[5].clone();
-
-    out[(0, 3)] = -m[4].clone() * m[9].clone() * m[14].clone()
-        + m[4].clone() * m[10].clone() * m[13].clone()
-        + m[8].clone() * m[5].clone() * m[14].clone()
-        - m[8].clone() * m[6].clone() * m[13].clone()
-        - m[12].clone() * m[5].clone() * m[10].clone()
-        + m[12].clone() * m[6].clone() * m[9].clone();
-
-    out[(3, 2)] = -m[0].clone() * m[5].clone() * m[11].clone()
-        + m[0].clone() * m[7].clone() * m[9].clone()
-        + m[4].clone() * m[1].clone() * m[11].clone()
-        - m[4].clone() * m[3].clone() * m[9].clone()
-        - m[8].clone() * m[1].clone() * m[7].clone()
-        + m[8].clone() * m[3].clone() * m[5].clone();
-
-    out[(1, 3)] = m[0].clone() * m[9].clone() * m[14].clone()
-        - m[0].clone() * m[10].clone() * m[13].clone()
-        - m[8].clone() * m[1].clone() * m[14].clone()
-        + m[8].clone() * m[2].clone() * m[13].clone()
-        + m[12].clone() * m[1].clone() * m[10].clone()
-        - m[12].clone() * m[2].clone() * m[9].clone();
-
-    out[(2, 3)] = -m[0].clone() * m[5].clone() * m[14].clone()
-        + m[0].clone() * m[6].clone() * m[13].clone()
-        + m[4].clone() * m[1].clone() * m[14].clone()
-        - m[4].clone() * m[2].clone() * m[13].clone()
-        - m[12].clone() * m[1].clone() * m[6].clone()
-        + m[12].clone() * m[2].clone() * m[5].clone();
-
-    out[(3, 3)] = m[0].clone() * m[5].clone() * m[10].clone()
-        - m[0].clone() * m[6].clone() * m[9].clone()
-        - m[4].clone() * m[1].clone() * m[10].clone()
-        + m[4].clone() * m[2].clone() * m[9].clone()
-        + m[8].clone() * m[1].clone() * m[6].clone()
-        - m[8].clone() * m[2].clone() * m[5].clone();
-
-    let det = m[0].clone() * out[(0, 0)].clone()
-        + m[1].clone() * out[(0, 1)].clone()
-        + m[2].clone() * out[(0, 2)].clone()
-        + m[3].clone() * out[(0, 3)].clone();
-
-    if !det.is_zero() {
+    let det = m[0].clone()
```

Comment:
This seems to be adding a bunch of additional multiplies compared to the previous version. Is there a way to reuse intermediate results here?